src/services/api.ts

```diff
@@ -100,8 +103,39 @@ export const setApi = (api: ApiService) => {
   apiValue = api;
 };
 
+/**
+ * @deprecated
+ */
 export const api = () => apiValue;
 
+/**
+ * hook to access the API service
+ * @returns
+ */
+export function useAPI(): () => Promise<ApiService> {
+  const apiKey = useApiKey();
+
+  const getAPI = useCallback(async () => {
```

Comment:
Maybe I'm missing something but I don't understand why this needs to be async, nothing inside of this function is async, and `useAPI` becomes much harder to use if it returns a promise instead of just returning an ApiService